src/worker.js

```diff
@@ -25,30 +25,50 @@ editModule("fs", (fs) => {
 
 const componentCompiler = require("@vue/component-compiler");
 
-module.exports = async ({ filename, source, trackUsedFiles }) => {
-  const compiler = componentCompiler.createDefaultCompiler();
+module.exports = async ({
+  filename,
+  source,
+  trackUsedFiles,
+  extractCss,
+  production,
+}) => {
+  const compilerOptions = production
+    ? { template: { isProduction: true } }
+    : {};
+  const compiler = componentCompiler.createDefaultCompiler(compilerOptions);
   usedFiles = new Set();
   try {
     if (/^\s*$/.test(source)) {
       throw new Error("File is empty");
     }
     const result = compiler.compileToDescriptor(filename, source);
+    let styles = [];
+
     const resultErrors = combineErrors(result.template, ...result.styles);
     if (resultErrors.length > 0) {
-      return { result: { errors: resultErrors }, usedFiles };
+      return { errors: resultErrors, usedFiles };
+    }
+    if (extractCss) {
+      styles = result.styles.map(({ code, scoped, media, module }) => ({
+        code,
+        scoped,
+        media,
+        module,
+      }));
```

Comment:
I think we need only `code`? In that case can both object patterns be changed to `{code}`?

Reply:
At the moment this PR support only global and scoped CSS but I think that the `media` and `module` property might be used when setting media query for the style block and CSS Modules. I have not tested those two scenarios though.

Reply:
I suppose the media query on the style block might be translated to a `@media ... { ... }` block wrapped around the exported CSS.

I'm not familiar with CSS modules and how they're used in inline style blocks in Vue.